FlightTelemetry_2_1_2020/FlightTelemetry_2_1_2020.ino

```diff
@@ -0,0 +1,150 @@
+RTC_DS1307 rtc;
+File dataFile;
+char filename[] = "DATA000.csv";
+
+#define LED    13// BUILTIN_SDCARD  //use BUILTIN_SDCARD for Teensy 3.6
+//also change line ~46 to: 
+//if (!SD.begin(BUILTIN_SDCARD)) { TELEMETRY_SERIAL.println(F("SD err")); }
+//10 (or other) for all other boards)
+
+//I'm not sure if the above is correct
+
+
+#define SEND_VECTOR_ITEM(field, value)\
+  SEND_ITEM(field, value.x())         \
+  SEND_GROUP_ITEM(value.y())          \
+  SEND_GROUP_ITEM(value.z())
+
+#define WRITE_CSV_ITEM(value)         \
+  dataFile.print(F(", ")); dataFile.print(value);
+
+#define WRITE_CSV_VECTOR_ITEM(value)  \
+  WRITE_CSV_ITEM(value.x())           \
+  WRITE_CSV_ITEM(value.y())           \
+  WRITE_CSV_ITEM(value.z())
+
+unsigned int missed_deadlines = 0;
+
+
+void setup() {
+  TELEMETRY_SERIAL.begin(57600);    TELEMETRY_SERIAL.println();
+
+  while (!bmp.begin()) {
+    TELEMETRY_SERIAL.println(F("BMP388 err"));
+    digitalWrite(LED,LOW); delay(1000); digitalWrite(LED,HIGH);
+  } 
+  
+  //does it make sense that the CS pin for the SD card is flashing on/off????
+
+  bmp.setTemperatureOversampling(BMP3_OVERSAMPLING_8X);
+  bmp.setPressureOversampling(BMP3_OVERSAMPLING_4X);
+  bmp.setIIRFilterCoeff(BMP3_IIR_FILTER_COEFF_3);
+
+  //put GPS initialization code here:
+
+
+
+  
+  
+  while (!bno.begin()) {                            // flashes to signal error
+    TELEMETRY_SERIAL.println(F("BNO055 err"));
+    digitalWrite(LED,LOW); delay(1000); digitalWrite(LED,HIGH);
+  }
+  if (!rtc.isrunning()) { rtc.adjust(DateTime(__DATE__, __TIME__)); }
+  if (!SD.begin(BUILTIN_SDCARD)) { TELEMETRY_SERIAL.println(F("SD err")); }
+  else {                                            // generates file name
+    for (uint16_t nameCount = 0; nameCount < 1000; nameCount++) {
+      filename[4] = nameCount/100 + '0';
+      filename[5] = (nameCount%100)/10 + '0';
+      filename[6] = nameCount%10 + '0';
+      if (!SD.exists(filename)) {                   // opens if file doesn't exist
+        dataFile = SD.open(filename, FILE_WRITE);
+        TELEMETRY_SERIAL.print(F("\twriting "));
+        TELEMETRY_SERIAL.println(filename);
+        dataFile.println(F("abs time,sys date,sys time,x_euler_angle,y_euler_angle,z_euler_angle,x_gyro,y_gyro,z_gyro,temperature,x_magnetometer,y_magnetometer,z_magnetometer,x_acceleration,y_acceleration,z_acceleration"));
+        break;
+      }
+    }
+  }
+  
+}
+
+
+void loop() {
+  long time0 = millis();
+  imu::Vector<3> gyroscope     = bno.getVector(Adafruit_BNO055::VECTOR_GYROSCOPE);
+  imu::Vector<3> euler         = bno.getVector(Adafruit_BNO055::VECTOR_EULER);
+  imu::Vector<3> accelerometer = bno.getVector(Adafruit_BNO055::VECTOR_ACCELEROMETER);
+  imu::Vector<3> magnetometer  = bno.getVector(Adafruit_BNO055::VECTOR_MAGNETOMETER);
+  int8_t temp = bno.getTemp();
+  
+  int8_t BMPtemp= bmp.temperature; //in C, do I need int8_t type????

```

Comment:
Style note: variables should be named lowercase.

Reply:
As for the type, you need to check the library.  In this case it appears they should be `float`s (or `double`s), otherwise you are losing a lot of precision.
https://github.com/adafruit/Adafruit_BMP3XX/blob/master/Adafruit_BMP3XX.h

Reply:
does choosing one or the other take up more bandwidth in the transmitting part?

Reply:
It gets sent as text right now, so only marginally.  I've been meaning to redo the packet wire format to be more compact, but haven't gotten around to it yet.